plugins/inputs/metric_streams_listener/README.md

```diff
@@ -0,0 +1,132 @@
+# Metric Streams Input Plugin
+
+The Metric Streams plugin is a service input plugin that
+listens for metrics sent via HTTP and performs the required
+processing for
+[Metric Streams from AWS](#troubleshooting-documentation).
+
+Metrics from Metric Streams are Base64 encoded JSON
+blocks that adhere to
+[AWS's request & response specifications](#troubleshooting-documentation).
+
+The Telegraf plugin must return the required response format
+as outlined in
+[the Firehose documentation](#troubleshooting-documentation).
+
+For cost, see the Metric Streams example in
+[CloudWatch pricing](#troubleshooting-documentation).
+
+## Configuration
+
+```toml @sample.conf
+[[inputs.metric_streams]]
+  ## Address and port to host HTTP listener on
+  service_address = ":443"
+
+  ## Paths to listen to.
+  # paths = ["/telegraf"]
+
+  ## maximum duration before timing out read of the request
+  # read_timeout = "10s"
+
+  ## maximum duration before timing out write of the response
+  # write_timeout = "10s"
+
+  ## Maximum allowed http request body size in bytes.
+  ## 0 means to use the default of 524,288,000 bytes (500 mebibytes)
+  # max_body_size = "500MB"
+
+  ## Optional username and password to accept for HTTP basic authentication.
+  ## You probably want to make sure you have TLS configured above for this.
+  # basic_username = "foobar"
+  # basic_password = "barfoo"
+```
+
+## Metrics
+
+Metrics sent by AWS are Base64 encoded blocks of JSON data.
+There can be multiple blocks of JSON for each `data` field
+in a `record` and there can be multiple `records`.
+
+The metric when decoded may look like this:
+
+```json
+{
+    "metric_stream_name": "sandbox-dev-cloudwatch-metric-stream",
+    "account_id": "541737779709",
+    "region": "us-west-2",
+    "namespace": "AWS/EC2",
+    "metric_name": "CPUUtilization",
+    "dimensions": {
+        "InstanceId": "i-0efc7ghy09c123428"
+    },
+    "timestamp": 1651679580000,
+    "value": {
+        "max": 10.011666666666667,
+        "min": 10.011666666666667,
+        "sum": 10.011666666666667,
+        "count": 1
+    },
+    "unit": "Percent"
+}
+```
+
+### Tags
+
+All tags in the `dimensions` list are added as tags to the metric.
+
+The `account_id` and `region` tag are added to each metric as well.
+
+### Measurements and Fields
+
+The metric name is a combination of `namespace` and `metric_name`,
+separated by `_` and lowercased.
+
+The fields are each aggregate in the `value` list.
+
+The timestamp applied is the timestamp from the metric,
+typically 3-5 minutes older than the time processed due
+to CloudWatch delays.
```

Comment:
Are the metrics produced by this plugin and the cloudwatch plugin the same? It would be a big usability win if they were because could switch from the cloudwatch plugin to this one without having to handle data model changes.

Reply:
We actually discussed this internally a bit, and we agree - we were going to make breaking changes in our own dashboards and alerts by conforming to the new naming after the migration from the CloudWatch API (pull method).

So I think you're right, and we should make this easier for anyone else who upgrades, so that we get that usability win you mentioned. So I will push an update to make the two align.

Reply:
That sounds great. I was in the middle of writing down my thoughts about it when you posted. Here it is anyway:

I think tweaking the names to match cloudwatch is the way to go if there are only a few differences.

If we don't make them compatible now, I expect that in the future someone will submit a PR to make them compatible. If we add compatibility at a future time, telegraf will need to make it optional so we don't break people who got used to the differences. Then the plugin will be stuck with maintaining two metric formats or face breaking compatibility for a portion of users.

If we do make them compatible now, we don't have to deal with two metric formats or breaking compatibility.

Reply:
Oh, missed this.

@reimda what about users not used to the other plugin, wouldn't they find it weird the names from metric stream get renamed here? IMHO it could be a config flag to choose..

Reply:
You have a good point. Let's use the same format as the other plugin for this PR and if users request support for the native metric streams naming we can look into adding a setting in another PR.

(I hope we don't end up making both plugins compatible with both naming variants in the future.)